packages/miniflare/test/plugins/kv/index.spec.ts

```diff
@@ -179,6 +179,17 @@ test("put: keys are case-sensitive", async (t) => {
 	result = await kv.get("KEY");
 	t.is(result, "upper");
 });
+// this test might look odd, but this is to verify a bug fix
+// where inserting numeric key was deleting preceeding entry.
+// "serial" is required to simulate the reproduction
+test.serial("put: key starting with number", async (t) => {
+	const { kv } = t.context;
+	await kv.put("abc-key", "abc-value");
+	t.is(await kv.get("abc-key"), "abc-value");
+	await kv.put("123-key", "123-value");
+	t.is(await kv.get("123-key"), "123-value");
+	t.is(await kv.get("abc-key"), "abc-value"); // this was becoming `null` previously
+});
```

Comment:
For now, I added a test to simulate the exact scenario of the reproduction, but the same thing can be actually tested by the existing test since they already have a numeric prefix due to namespacing.
I think this bug was somehow not surfaced because all tests are run in parallel.

For example, I was able to see this existing test to fail when running with `test.only` with `ava test/plugins/kv/index.spec.ts`:

https://github.com/cloudflare/workers-sdk/blob/6c7e2796b39642c767665ccdb432b13da3281967/packages/miniflare/test/plugins/kv/index.spec.ts#L173-L181

```
  put: keys are case-sensitive

  test/plugins/kv/index.spec.ts:178

   177:   let result = await kv.get("key");
   178:   t.is(result, "lower");           
   179:   result = await kv.get("KEY");    

  Difference (- actual, + expected):

  - null
  + 'lower'
```